cargo msrv test is failing on main for `object_store`

**Describe the bug**
Example:
https://github.com/apache/arrow-rs/actions/runs/9454737839/job/26042972148

```
Run cargo msrv verify
Fetching index
Verifying the Minimum Supported Rust Version (MSRV) for toolchain x[8](https://github.com/apache/arrow-rs/actions/runs/9454737839/job/26042972148#step:11:9)6_64-unknown-linux-gnu
Using check command cargo check
   Failed check command cargo check didn't succeed
Crate source was found to be incompatible with its MSRV '1.62.1', as defined in '/__w/arrow-rs/arrow-rs/object_store/Cargo.toml'
Error: Process completed with exit code 1.
```

**To Reproduce**
Run CI on `object_store`

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

**Additional context**
<!--
Add any other context about the problem here.
-->

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -132,7 +132,7 @@ jobs:
         # and url 2.5.1, updates to 1.67
         run: |
           cargo update -p tokio --precise 1.29.1
-          cargo update -p url --precise 2.5.1
+          cargo update -p url --precise 2.5.0
       - name: Check object_store
         working-directory: object_store
         run: cargo msrv --log-target stdout verify

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -119,17 +119,20 @@ jobs:
         run: cargo update -p ahash --precise 0.8.7
       - name: Check arrow
         working-directory: arrow
-        run: cargo msrv verify
+        run: cargo msrv --log-target stdout verify
       - name: Check parquet
         working-directory: parquet
-        run: cargo msrv verify
+        run: cargo msrv --log-target stdout verify
       - name: Check arrow-flight
         working-directory: arrow-flight
-        run: cargo msrv verify
+        run: cargo msrv --log-target stdout verify
       - name: Downgrade object_store dependencies
         working-directory: object_store
-        # Necessary because 1.30.0 updates MSRV to 1.63
-        run: cargo update -p tokio --precise 1.29.1
+        # Necessary because tokio 1.30.0 updates MSRV to 1.63
+        # and url 2.5.1, updates to 1.67
+        run: |
+          cargo update -p tokio --precise 1.29.1
+          cargo update -p url --precise 2.5.1
       - name: Check object_store
         working-directory: object_store
-        run: cargo msrv verify
+        run: cargo msrv --log-target stdout verify

